### PR TITLE
Add Strategy page to docs

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Part of #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/extensions/strategy.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Training Strategies with various configs
+****************************************
+
+.. code-block:: python
+
+    # Training with the DistributedDataParallel strategy on 4 gpus
+    trainer = Trainer(strategy="ddp", accelerator="gpu", devices=4)
+
+    # Training with the custom DistributedDataParallel strategy on 4 gpus
+    trainer = Trainer(strategy=DDPPlugin(...), accelerator="gpu", devices=4)
```
Comment (ananthsub) [inline docs/source/extensions/strategy.rst:20]:
```suggestion
    trainer = Trainer(strategy=DDPStrategy(...), accelerator="gpu", devices=4)
```

File: docs/source/extensions/strategy.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
+.. _strategy:
+
+########
+Strategy
+########
+
+Strategy depicts the training strategy to be used by the :doc:`Lightning Trainer <../common/trainer>`. It can be controlled by passing different
```
Comment (ananthsub) [inline docs/source/extensions/strategy.rst:7]: Strategy depicts training strategy is circular.

We should define what strategy is meant to encompass, and only afterward list out how someone can use it. For instance, Strategy controls the model distribution across training, evaluation, and prediction.

File: docs/source/extensions/strategy.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    trainer = Trainer(strategy="ddp", accelerator="cpu", devices=3)
+
+    # Training with the DDP Spawn strategy on 8 tpu cores
+    trainer = Trainer(strategy="ddp_spawn", accelerator="tpu", devices=8)
+
+    # Training with the default IPU strategy on 8 ipus
+    trainer = Trainer(accelerator="ipu", devices=8)
+
+
+Strategy in Lightning handles some of the following responsibilities:
```
Comment (ananthsub) [inline docs/source/extensions/strategy.rst:38]: This should go earlier imo.

Comment (kaushikb11) [reply docs/source/extensions/strategy.rst]: Updated!

File: docs/source/extensions/strategy.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source/extensions/strategy.rst:7]: I would agree, strategy does not fully fit into what extensions are. it is more an Option, as in, many different strategies are available to choose from but it is required to have one.

File: docs/source/extensions/strategy.rst [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+* Provide access to the wrapped LightningModule
+
+
+:class:`~pytorch_lightning.strategies.strategy.Strategy` also manages the accelerator, precision and the checkpointing plugins.
+
+
+****************************************
+Training Strategies with various configs
+****************************************
```
Comment (awaelchli) [inline docs/source/extensions/strategy.rst:26]: @rohitgr7 has started to standardize all titles to be capitalized

File: docs/source/extensions/strategy.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source/extensions/strategy.rst:7]: Strategies can also be "extended", which is probably why @kaushikb11 has put it here, but this is definitely a different category than Callbacks for example.

File: docs/source/extensions/strategy.rst [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+Strategy in Lightning handles some of the following responsibilities:
+
+* Launching and teardown of training processes (if applicable)
+
+* Setup communication between processes (NCCL, GLOO, MPI, …)
+
+* Provide a unified communication interface for reduction, broadcast, etc.
+
+* Provide access to the wrapped LightningModule
```
Comment (awaelchli) [inline docs/source/extensions/strategy.rst:18]: I would even just say that the strategy owns the LightningModule

File: docs/source/extensions/strategy.rst [hunk 6]
```diff
@@ -0,0 +1,111 @@
+.. _strategy:
+
+########
+Strategy
+########
+
+Strategy depicts the training strategy to be used by the :doc:`Lightning Trainer <../common/trainer>`. It can be controlled by passing different
+training strategies with aliases (``ddp``, ``ddp_spawn``, ``deepspeed``, etc) as well as custom training strategies to the ``strategy`` parameter for Trainer.
+
+Strategy in Lightning handles some of the following responsibilities:
```
Comment (awaelchli) [inline docs/source/extensions/strategy.rst:10]: Can also mention the responsibility of setting up / owning of optimizers, lrschedulers?

File: docs/source/extensions/strategy.rst [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+
+    class CustomDDPStrategy(DDPStrategy):
+        def configure_ddp(self):
+            self._model = MyCustomDistributedDataParallel(
+                self.model,
+                device_ids=...,
+            )
+
+or by subclassing the base classes :class:`~pytorch_lightning.strategies.Strategy` to create new ones. These custom strategies
```
Comment (awaelchli) [inline docs/source/extensions/strategy.rst:70]:
```suggestion
or by subclassing the base class :class:`~pytorch_lightning.strategies.Strategy` to create new ones. These custom strategies
```

File: docs/source/extensions/strategy.rst (hunk 6 above)
Comment (awaelchli) [inline docs/source/extensions/strategy.rst:10]:
```suggestion
The Strategy in Lightning handles the following responsibilities:
```
If you wrote "some of the responsibilities" that would mean only a subset of the below listed items are handled, which is not what you mean.

**Review by awaelchli — APPROVED** [review]
Good idea to move this out of the plugins page. Intro sentence could be improved

File: docs/source/extensions/strategy.rst (hunk 2 above)
Comment (tchaton) [inline docs/source/extensions/strategy.rst:7]: Furthermore, we renamed them strategy as they were not related to training only. This invalidates this change ahah

File: docs/source/extensions/strategy.rst (hunk 6 above)
Comment (tchaton) [inline docs/source/extensions/strategy.rst:10]: I think a gif would make the explanation much simpler.

File: docs/source/extensions/strategy.rst [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+Expert users may choose to extend an existing strategy by overriding its methods ...
+
+.. code-block:: python
+
+    from pytorch_lightning.strategies import DDPStrategy
+
+
+    class CustomDDPStrategy(DDPStrategy):
+        def configure_ddp(self):
+            self._model = MyCustomDistributedDataParallel(
```
Comment (tchaton) [inline docs/source/extensions/strategy.rst:65]: Usually custom DDP will come with their own ClusterEnv such as Bagua. Worth mentioning this.

File: docs/source/extensions/strategy.rst [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+.. code-block:: python
+
+    # custom plugins
+    trainer = Trainer(strategy=CustomDDPStrategy())
+
+    # fully custom accelerator and plugins
+    accelerator = MyAccelerator()
+    precision_plugin = MyPrecisionPlugin()
+    training_strategy = CustomDDPStrategy(accelerator=accelerator, precision_plugin=precision_plugin)
```
Comment (tchaton) [inline docs/source/extensions/strategy.rst:81]: Let's add hybrid accelerator mechanism that can be implemented by the user if needed.

File: docs/source/extensions/strategy.rst (hunk 2 above)
Comment (kaushikb11) [reply docs/source/extensions/strategy.rst]: Resolved!

File: docs/source/extensions/strategy.rst (hunk 6 above)
Comment (kaushikb11) [reply docs/source/extensions/strategy.rst]: I agree! @rohitgr7 I think we could audit the docs again where visual content could benefit.

File: docs/source/extensions/strategy.rst (hunk 9 above)
Comment (kaushikb11) [reply docs/source/extensions/strategy.rst:82]: What do you mean by hybrid here?

**Review by ananthsub — APPROVED** [review]

File: docs/source/extensions/strategy.rst [hunk 10]
```diff
@@ -0,0 +1,116 @@
+.. _strategy:
+
+########
+Strategy
+########
+
+Strategy controls the model distribution across training, evaluation, and prediction to be used by the :doc:`Trainer <../common/trainer>`. It can be controlled by passing different
+strategy with aliases (``"ddp"``, ``"ddp_spawn"``, ``"deepspeed"`` and so on.) as well as custom strategy to the ``strategy`` parameter for Trainer.
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:8]:
```suggestion
strategy with aliases (``"ddp"``, ``"ddp_spawn"``, ``"deepspeed"`` and so on) as well as a custom strategy to the ``strategy`` parameter for Trainer.
```
Probably?

File: docs/source/extensions/strategy.rst [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+The Strategy in Lightning handles the following responsibilities:
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:10]:
```suggestion
The Strategy in PyTorch Lightning handles the following responsibilities:
```

File: docs/source/extensions/strategy.rst [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+* Launching and teardown of training processes (if applicable)
+* Setup communication between processes (NCCL, GLOO, MPI and so on.)
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:13]:
```suggestion
* Setup communication between processes (NCCL, GLOO, MPI, and so on).
```

File: docs/source/extensions/strategy.rst [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+########
+Strategy
+########
+
+Strategy controls the model distribution across training, evaluation, and prediction to be used by the :doc:`Trainer <../common/trainer>`. It can be controlled by passing different
+strategy with aliases (``"ddp"``, ``"ddp_spawn"``, ``"deepspeed"`` and so on.) as well as custom strategy to the ``strategy`` parameter for Trainer.
+
+The Strategy in Lightning handles the following responsibilities:
+
+* Launching and teardown of training processes (if applicable)
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:12]:
```suggestion
* Launch and teardown of training processes (if applicable).
```

File: docs/source/extensions/strategy.rst [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+* Setup communication between processes (NCCL, GLOO, MPI and so on.)
+* Provide a unified communication interface for reduction, broadcast and so on.
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:14]:
```suggestion
* Provide a unified communication interface for reduction, broadcast, and so on.
```

File: docs/source/extensions/strategy.rst [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+* Owns the LightningModule
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:15]:
```suggestion
* Owns the `LightningModule`.
```
Or should this be:

```
:class:`~pytorch_lightning.LightningModule`
```

File: docs/source/extensions/strategy.rst [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+* Handles/owns optimizers and schedulers
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:16]:
```suggestion
* Handles/owns optimizers and schedulers.
```

File: docs/source/extensions/strategy.rst [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+
+
+:class:`~pytorch_lightning.strategies.strategy.Strategy` also manages the accelerator, precision and the checkpointing plugins.
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:19]:
```suggestion
:class:`~pytorch_lightning.strategies.strategy.Strategy` also manages the accelerator, precision, and checkpointing plugins.
```

File: docs/source/extensions/strategy.rst [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+    class CustomDDPStrategy(DDPStrategy):
+        def configure_ddp(self):
+            self.model = MyCustomDistributedDataParallel(
+                self.model,
+                device_ids=...,
+            )
+
+or by subclassing the base class :class:`~pytorch_lightning.strategies.Strategy` to create new ones. These custom strategies
+can then be passed into the Trainer directly via the ``strategy`` parameter.
```
Comment (krshrimali) [inline docs/source/extensions/strategy.rst:72]:
```suggestion
can then be passed into the ``Trainer`` directly via the ``strategy`` parameter.
```

Probably?

**Review by krshrimali — APPROVED** [review]
Nice, LGTM 🚀 Left very minor comments (mostly through Grammarly) for your reference. Feel free to ignore some if they look irrelevant. :)

File: docs/source/extensions/strategy.rst (hunk 15 above)
Comment (rohitgr7) [reply docs/source/extensions/strategy.rst:15]:
```
:class:`~pytorch_lightning.core.lightning.LightningModule`
```

**Comment by daniellepintz** [comment]
Shouldn't Strategy be removed from here as well? https://pytorch-lightning.readthedocs.io/en/latest/extensions/plugins.html#strategy

**Comment by rohitgr7** [comment]
> Shouldn't Strategy be removed from here as well? https://pytorch-lightning.readthedocs.io/en/latest/extensions/plugins.html#strategy

cc @kaushikb11
